taskboot/target.py

```diff
@@ -35,27 +37,16 @@ def clone(self, repository, revision):
         logger.info("Cloning {} @ {}".format(repository, revision))
 
         # Clone
-        cmd = ["git", "clone", "--quiet", repository, self.dir]
+        cmd = ["git", "-c", "init.defaultBranch=clone", "init", self.dir]
         subprocess.check_output(cmd)
+        cmd = ["git", "remote", "add", "origin", repository]
+        subprocess.check_output(cmd, cwd=self.dir)
+        cmd = ["git", "fetch", "--quiet", "origin", revision]
+        retry(lambda: subprocess.check_output(cmd, cwd=self.dir))
         logger.info("Cloned into {}".format(self.dir))
 
-        # Explicitly fetch revision if it isn't present
-        # This is necessary when revision is from a fork
-        # and repository is the base repo.
-        if (
-            subprocess.run(
-                ["git", "show", revision],
-                cwd=self.dir,
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
-            ).returncode
-            != 0
-        ):
-            cmd = ["git", "fetch", "--quiet", "origin", revision]
-            subprocess.check_output(cmd, cwd=self.dir)
-
         # Checkout revision to pull modifications
-        cmd = ["git", "checkout", revision, "-b", "taskboot"]
+        cmd = ["git", "checkout", "FETCH_HEAD", "-b", "taskboot"]
```

Comment:
Nit: is this change necessary?

Reply:
It is. Without this, only revision hashes work, not names.

```
>>> from taskboot.target import Target
>>> from argparse import Namespace
>>> args = Namespace(git_repository="https://github.com/MozillaSecurity/orion", git_revision="master", target=None)
>>> Target(args)
fatal: 'master' is not a commit and a branch 'taskboot' cannot be created from it
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/truber/src/m/task-boot/taskboot/target.py", line 32, in __init__
    self.clone(args.git_repository, args.git_revision)
  File "/home/truber/src/m/task-boot/taskboot/target.py", line 50, in clone
    subprocess.check_output(cmd, cwd=self.dir)
  File "/usr/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'checkout', 'master', '-b', 'taskboot']' returned non-zero exit status 128.
```

Using `FETCH_HEAD`, both `master` or a git revision hash will work.